### PR TITLE
Rk 624 add suggests

### PR DESCRIPTION
**Pull request by Robert-Krajcik**

### CODE REVIEW

File: inst/sql_queries/initialize_metric_table.sql [hunk 1]
```diff
@@ -14,4 +14,5 @@ VALUES
 ('dependencies',        'Dependencies',      'Number of Package Dependencies',   0, 0, 'maintenance', 1),  
 ('reverse_dependencies','Reverse Dependencies', 'Number of Reverse Dependencies', 0, 0, 'community', 1),
 ('covr_coverage',       'Test Coverage',     '% of objects tested',     1, 0, 'maintenance', 1),
-('downloads_1yr', 'Package Downloads', 'Number of downloads in last 12 months', 0, 0, 'community', 1);
\ No newline at end of file
+('downloads_1yr', 'Package Downloads', 'Number of downloads in last 12 months', 0, 0, 'community', 1),
+('suggests', 'Suggests', 'Number of Package Suggests', 0, 0, 'dependency', 0);
```
Comment (AARON-CLARK) [inline inst/sql_queries/initialize_metric_table.sql:18]: Note: whenever you change the database in any way, you now need to run `creating_testing_databases.R` in the `data-raw` folder thanks to #629 . But that is a pretty recent change so I wouldn't expect you to be aware of it yet. Jeff did add a new bullet point to the [PR checklist](https://github.com/pharmaR/riskassessment/wiki/Developer-Pull-Request-workflows#pr-checklist), but @Jeff-Thompson12, could you also announce this new workflow and give a quick recap of what it does on slack so everyone is aware?

Comment (AARON-CLARK) [reply inst/sql_queries/initialize_metric_table.sql:18]: Also, I think this is the first non `riskmetric` metric we are actually storing in the database. All the other ones included in the app are calculated on the fly.

**Review by AARON-CLARK — CHANGES_REQUESTED** [review]
Hi @Robert-Krajcik, thanks for this PR! I think it will be valuable for users to see the `Suggests` pkg lists in the app. I have a few comments:

First, the button label "Hide Suggests" is wrapping for me at 100% zoom in the browser. If I reduce my zoom to 80%, it appears on one line and becomes visible again. Also, I don't think we should adopt a button in this case since it's hard to tell when the suggests pkgs are hidden vs visible, unless you inspect the table. It's even harder to tell because the `DT` default is to just show the top 15 pkgs, and the suggests may not show up in the top rows. I sorted the table so mine show up at the top. Therefore, I'd switch over to using `shinyWidgets::materialSwitch()` to solve both problems - you'll see the text next to the toggle and it will be obvious when it's "on" vs "off".

image

Or, alternatively, you could make each type into a "select button", like we did for `tidyCDISC`. That approach could be useful if users only want to skip more than just suggests. Here is the link to the code:

https://github.com/Biogen-Inc/tidyCDISC/blob/f41ed77e1fd0df86ad95ec75dbb01efa565d00af/R/mod_dataUpload.R#L25-L31

Also looking at the image above, I wonder if people would know that the 3 suggests are not counted in the 25 dependencies. I think this will be more obvious when we move to work on #618, which will show a breakdown of each group/type.

File: inst/sql_queries/initialize_metric_table.sql (hunk 1 above)
Comment (Robert-Krajcik) [reply inst/sql_queries/initialize_metric_table.sql:18]: OK
added the materialSwitch in place of an action button
and ran `creating_testing_databases.R`

**Comment by Robert-Krajcik** [comment]
Need to checkout a quirk in the `hide_suggests` code

**Review by AARON-CLARK — CHANGES_REQUESTED** [review]
> Using reactiveVal toggled() to prevent materialSwitch from resetting itself to FALSE every time.

Hi @Robert-Krajcik, it looks like this isn't fixed yet. It's still resetting for me every time I change a package in the control panel, or via the --> buttons. Could you look into this & verify? Thanks!

**Comment by Robert-Krajcik** [comment]
It seems to be working for me, @AARON-CLARK 
Maybe I'll build a reprex.

**Comment by AARON-CLARK** [comment]
@Robert-Krajcik, I deleted the db & tried restarting and terminating R. Plus, I verified that I have all the latest commits:

image

And this is what I'm seeing:

image

image

**Comment by Robert-Krajcik** [comment]
Hi @AARON-CLARK 
This is ready for another review.

The only caveat I have is that packages like `{magrittr}` , `{curl}` and `{lubridate}` 
need to have the `hide_suggests` switch set to FALSE (I set `toggle(0L)` in those cases where nrow(depends()) == 0)
So if you were looking at `{forcats}` with the `hide_suggests` switch set to TRUE
and then select `{magrittr}`, the next package selected will have `hide_suggests` set to FALSE

Getting it to remember to previous `toggle()` value might take some time, but I think this is pretty good as is.

**Comment by AARON-CLARK** [comment]
> The only caveat I have is that packages like `{magrittr}` , `{curl}` and `{lubridate}`
> need to have the `hide_suggests` switch set to FALSE (I set `toggle(0L)` in those cases where nrow(depends()) == 0)

Can you **not** manually override the `hide_suggests` toggle to zero for `{magrittr}` , `{curl}` and `{lubridate}` (or other pkgs where nrow(depends()) == 0)? In other words, keep things AS-IS by by displaying no table?

image

**Comment by Robert-Krajcik** [comment]
I **could** do that, but that would hide the suggests packages, which I think we would want to show.
`{magrittr}` has zero dependencies, **but** it has five suggests.
image

**Comment by AARON-CLARK** [comment]
> I **could** do that, but that would hide the suggests packages

I would do it! If the user is intentionally hiding the Suggests pkgs and there are no other pkg dependencies, the table should be empty. Remember, we'll be working on #618 next, so that will show a breakdown of the pkgs by TYPE. So, once that PR is developed, even though the table is empty, the summary cards at the top will display "0 Dependencies" but "5 Suggests"... so the user will know what's going on.

**Comment by Robert-Krajcik** [comment]
And if the user is **not** intentionally hiding the Suggests pkgs

**Comment by AARON-CLARK** [comment]
> And if the user is **not** intentionally hiding the Suggests pkgs

Yes, if they are not choosing to hide Suggests, we should show Suggests.

**Comment by Robert-Krajcik** [comment]
If the user chooses to select hide_suggests, and there is data, I am showing "no data available in table"
as there really is data there; but they chose not to view it.
image

**Review by AARON-CLARK — APPROVED** [review]
💪🏼

**Pull request merged**